examples/example-hono-api/src/auth0.ts

```diff
@@ -0,0 +1,175 @@
+import type { Context } from 'hono';
+import { env } from 'hono/adapter';
+import type { MiddlewareHandler } from 'hono/types';
+import { ApiClient } from '@auth0/auth0-api-js';
+import { HTTPException } from 'hono/http-exception';
+import { JWTPayload } from 'jose';
+
+export type Auth0JwtEnv = {
+  Variables: {
+    jwtPayload: JWTPayload
+  }
+};
+
+export const jwt = (
+  options?: {
+    auth0_domain: string;
+    auth0_audience: string;
+  },
+): MiddlewareHandler => {
+
+  let apiClient: ApiClient | null = null;
+
+  return async function jwt(ctx, next) {
+
+    // 1. Create an instance of ApiClient and pass it the configuration
+    if (!apiClient) {
+      const auth0Env = env(ctx);
+      const { auth0_domain, auth0_audience } = options || {
+        auth0_domain: auth0Env.AUTH0_DOMAIN,
+        auth0_audience: auth0Env.AUTH0_AUDIENCE,
+      };
+      if (!auth0_domain || auth0_domain.length === 0) {
+        throw new Error('JWT auth middleware requires options "auth0_domain"');
+      }
+      if (!auth0_audience || auth0_audience.length === 0) {
+        throw new Error('JWT auth middleware requires options "auth0_audience"');
+      }
+
+      apiClient = new ApiClient({
+        domain: auth0_domain,
+        audience: auth0_audience,
+      });
+    }
+
+    const credentials = ctx.req.raw.headers.get('Authorization');
+    if (!credentials) {
+      const errDescription = 'No Authorization header included in request';
+      throw new HTTPException(401, {
+        message: errDescription,
+        res: unauthorizedResponse({
+          ctx,
+          error: 'invalid_request',
+          errDescription,
+        }),
+      });
+    }
+
+    const parts = credentials.split(/\s+/);
+    if (parts.length !== 2) {
+      const errDescription = 'Invalid Authorization header structure';
+      throw new HTTPException(401, {
+        message: errDescription,
+        res: unauthorizedResponse({
+          ctx,
+          error: 'invalid_request',
+          errDescription,
+        }),
+      });
+    }
+
+    if (parts[0] !== 'Bearer') {
+      const errDescription =
+        'Invalid authorization header (only Bearer tokens are supported)';
+      throw new HTTPException(401, {
+        message: errDescription,
+        res: unauthorizedResponse({
+          ctx,
+          error: 'invalid_request',
+          errDescription: errDescription,
+        }),
+      });
+    }
+
+    const accessToken = parts[1];
+    if (!accessToken || accessToken.length === 0) {
+      const errDescription = 'No token included in request';
+      throw new HTTPException(401, {
+        message: errDescription,
+        res: unauthorizedResponse({
+          ctx,
```

Comment:
Do we need all of this for a sample application? I understand that, if we want to make it an actual SDK, we want to be more defensive and have clear errors.

But curious why we cant use what we use in other samples here?

```ts
const accessToken = ctx.req.raw.headers.get('Authorization')?.split(' ')[1];

      if (!accessToken) {
        return replyWithError(
          reply,
          400,
          'invalid_request',
          'No Authorization provided'
        );
      }
```

I feel like the code above unneccesary clutters the sample app, making it more complicated to parse.